doc/doxygen/src/driver-guide.md

```diff
@@ -0,0 +1,381 @@
+Writing a Device Driver in RIOT                                 {#driver-guide}
+===============================
+
+This document describes the requirement, design objectives, and some
+non-function details when writing device drivers in/for RIOT. The term device
+driver in this context includes all 'CPU-external' devices connected to the CPU
+typically via peripherals like SPI, I2C, UART, GPIO, and similar. CPU
+peripherals itself are in RIOT not considered to be device drivers, but
+peripheral or low-level drivers. Typical devices are network devices like
+radios, Ethernet adapters, sensors, and actuators.
+
+[TOC]
+
+# General Design Objectives                   {#driver-guide-design-objectives}
+
+Device drivers should be as easy to use as possible. This implies an
+'intialize->ready' paradigm, meaning, that device drivers should be ready to use
+and in operation right after they have been initialized. On top, devices should
+work with physical values where ever possible. So e.g. sensors should return
+already converted values in some physical unit instead of RAW data, so that
+users can work directly with data from different devices directly without having
+to care for device specific conversion.
+
+Additionally towards ease of use, all device drivers in RIOT should provide a
+similar 'look and feel'. They should behave similar concerning things like their
+state after initialization, like their used data representation and so on.
+
+Secondly, all device drivers should be optimized for minimal RAM/ROM usage, as
```

Comment:
I'm looking for "First" somewhere but can't find it. I think it should be at the beginning of one of the above paragraphs.

Reply:
It should. Again, I would like to keep this change out of this PR to keep this PR focus only on the move to the RIOT Repo